public/js/home.js

```diff
@@ -1181,6 +1192,220 @@ function closeSideNav() {
+        snackbarNode.id = `snackbar-${this.id}`;
+
+        //adds color if given
+        if (this.color !== undefined) {
+            snackbarNode.style.backgroundColor = `${this.color}`;
+        }
+
+        //sets the body onclick listener which just destroys it by default
+        const bodyOnClickFunction = this.bodyClick !== undefined ? () => {this.bodyClick();} : () => {};
+        const destroyFromBody = this.destroyWhenBodyClicked ? () => {this.destroy();} : () => {};
+        snackbarNode.addEventListener("click", () => {
+            bodyOnClickFunction();
+            destroyFromBody();
+        })
+
+        //adds the text
+        const textNode = document.createElement("SPAN");
+        textNode.textContent = this.text;
+
+        //colors the text if necessary
+        if (this.textColor !== undefined) {
+            textNode.style.color = this.textColor;
+        }
+
+        //adds the text node
+        snackbarNode.appendChild(textNode);
+
+        //makes the button if given button parameters
+        if (this.buttonText !== undefined && this.buttonClick != undefined) {
+
+            //creates the button and adds class
+            const buttonNode = document.createElement("BUTTON");
+
+            //creates the text span
+            const buttonTextNode = document.createElement("SPAN");
+            buttonTextNode.textContent = this.buttonText;
+
+            //colors the text if necessary
+            if (this.textColor !== undefined) {
+                buttonTextNode.style.color = this.textColor;
+            }
+
+            if (this.color !== undefined) {
+                buttonNode.style.backgroundColor = this.color;
+            }
+
+            //adds the text node
+            buttonNode.appendChild(buttonTextNode);
+
+            //sets the button onclick listener which runs the given funtion and destroys the snackar by default
+            const destroyFromButton = this.destroyWhenButtonClicked ? () => {this.destroy();} : () => {};
+            buttonNode.addEventListener("click", (event) => {
+                this.buttonClick();
+                destroyFromButton();
+                //stops propogation so the body event isn't called
+                event.stopPropagation();
+            })
+
+            snackbarNode.appendChild(buttonNode);
+        }
+
+        //adds the node to the body, puts reference to DOM element in this.element
+        document.body.appendChild(snackbarNode);
+        this.element = document.getElementById(`snackbar-${this.id}`);
+
+        return this;
+    }
+
+    /**
+     * shows the snackbar
+     * if it's not already made, makes it
+     * if you need to show right after making, use this
+     * returns the snackbar object
+     */
+    show() {
+        const snackbar = this;
+        
+        //function for the callback for setTimeout
+        const removeHidden = function() {
+            snackbar.element.classList.remove("hidden");
+        }
```

Comment:
I think this can be simplified while still paying attention to scope (untested):
```suggestion
        const removeHidden = () => this.element.classList.remove("hidden");
```